Show a feedback message for whispers you cannot hear

00-Steven opened the pull request:
## About The Pull Request

In this pr we make it so, instead of doing an early return, whispers you cannot hear will display a feedback message along the lines of:
`Angel Randolph says something, but you are too far away to understand him.`
As long as you are capable of seeing them say something.

This includes signing, as it uses the same logic.
If they're not signing and you cannot hear at all, then it skips to the normal deaf messages.


<details>
  <summary>Example Test Images</summary>

### explanation:

The number after each message denotes the distance between speaker and listener.
The first image is from the speaker's pov, the second from the listener's.
 
 
### no blindfold, no earmuffs:

no signing
![image](https://github.com/user-attachments/assets/eaf271e6-3174-40e5-8eec-647aba239e55)
![image](https://github.com/user-attachments/assets/272ef988-46ea-4c40-91a4-f55df2404d5b)

signing
![image](https://github.com/user-attachments/assets/9ef54ff9-9a63-4291-800b-9adce2e00727)
![image](https://github.com/user-attachments/assets/acd76a99-8395-4e10-90c4-dc28aaf3d888)


### no blindfold, earmuffs:

no signing
![image](https://github.com/user-attachments/assets/bf839a0e-8a0e-433c-b60e-d3e0677f6f39)
![image](https://github.com/user-attachments/assets/6da2e226-2149-45a5-9214-3e80bdd71c30)

signing
![image](https://github.com/user-attachments/assets/3fcf08ea-81fb-4e1f-a3d7-4037c9d75635)
![image](https://github.com/user-attachments/assets/e20a4e7b-4fcd-4f2c-9bee-b1bf8cdd9e5d)


### blindfold, no earmuffs:

no signing
![image](https://github.com/user-attachments/assets/963467b7-8cca-44d8-94f5-3f75d86f5e3a)
![image](https://github.com/user-attachments/assets/83793e34-b386-4595-8063-164915ff43e2)


signing
![image](https://github.com/user-attachments/assets/1a8f220d-02f2-4def-826a-a313a35f4894)
![image](https://github.com/user-attachments/assets/248c5e46-ed9d-4f87-b1da-ed6a361228e4)


### blindfold, earmuffs:

no signing
![image](https://github.com/user-attachments/assets/d084a21b-c050-44ba-b16a-86a7b3d2aa4c)
![image](https://github.com/user-attachments/assets/7b797034-7b0d-4238-b2eb-135193cab5c7)




</details>

## Why It's Good For The Game

Currently, someone whispering grants zero feedback to listeners who are beyond 2 tiles (without good hearing), all they get is that the speaker's typing indicator disappeared. This makes it look indistinguishable from the speaker just having backspaced their message.
This is _especially_ annoying with soft-spoken, as it tends to cause other players to disengage, because to them you just didn't press enter if they left your immediate range before you finished speaking.

But even outside of that, arguably, even if you couldn't _hear_ it, you should be able to _see_ the movements of them speaking.
And really, I just think that it'd lead to fun interactions if you could grumble to yourself or whisper amongst each other and have people actually know you're up to _something_.

## Changelog
:cl:
add: You can now see people whispering, even if you cannot hear what they're saying, unless you are blind (obviously). The speaker wearing something that covers their mouth, being invisible, or being inside of something counteracts this.
/:cl:

00-Steven committed:
see whispers

00-Steven committed:
oops all my tabs gone

scrunklershrungle932 commented:
does the message still show up if you're wearing anything that covers the mouth?

00-Steven commented:
I believe yes, currently, but so does the deaf `X says something but you cannot hear them` message.
I agree it probably _should_ block both of these, but I feel that's outside of the scope of this pr, given that'd require going through all our mask/helmet/neck slot items and working out which do and don't cover the mouth.

Sadboysuss commented:
We have a flag for `covers_mouth` or smth like that

00-Steven commented:
oh do we? let me take a look if that'd be helpful here, thank you

00-Steven commented:
Seems those'd be `MASKCOVERSMOUTH` and `HEADCOVERSMOUTH`, which while the overlap isn't perfect (I feel the default see-through breath mask probably shouldn't block it, for one) that may be Good Enough here, yeah

00-Steven commented:
like it seems it's more of a "is this physically blocked" than "is this _visually_ blocked", but it feels usable enough

snailomi commented:
This is a nice pr! a bit of a nitpick, but maybe it would be better to word "signs something, but you are too far away to understand them" as "signs something, but the motions are too subtle to make out" since the whisper signing is prefaced as "subtly signs". not at all important though, the current wording is perfectly fine!

00-Steven commented:
Considering it, but it does miss out on the distance aspect- I'll think about it a bit more after I think about implementing the face covering thing

Dorsisdwarf commented:
Seems like an annoying way to spread radio double-whispering and make it clog up the chatlog even more.

00-Steven commented:
I don't believe this would even apply to that, given it's entirely on the side of living entities hearing it.

snailomi commented:
> Considering it, but it does miss out on the distance aspect-

good point, maybe add on a "from afar" to the end to fix that?

Jackal-boop commented:
I think I don't like this. Whispering to an antag buddy when people are around is a really useful tool, this is gonna make people valid 2 people whispering in a corner somewhere I fear

00-Steven committed:
update signing feedback, account for more things

accounts for:
invisible (saturnx)
being in a closet/crate/such (notably includes stealth implant)
having your mouth covered (when not signing)

00-Steven commented:
Should now account for:
\- mouth being covered
\- being inside of something (cardboard box, closets, crates, stealth implant box)
\- saturnx invisibility
I feel further edge cases should probably be addressed separately

Nerev4r commented:
why not just 'whispers something' or smth

DATA-xPUNGED commented:
I also agree that the message maybe should just be "X whispers something." The one you placed is quite big

00-Steven committed:
maybe remember to actually commit and push

now it's roughly "X [verb_whisper] something, but you are too far away to hear [pronoun]."
